lib/core/dispatchRequest.js

```diff
@@ -35,41 +35,27 @@ module.exports = function dispatchRequest(config) {
     }
   );
 
-  var adapter;
-
-  if (typeof config.adapter === 'function') {
-    // For custom adapter support
-    adapter = config.adapter;
-  } else if (typeof XMLHttpRequest !== 'undefined') {
-    // For browsers use XHR adapter
-    adapter = require('../adapters/xhr');
-  } else if (typeof process !== 'undefined') {
-    // For node use HTTP adapter
-    adapter = require('../adapters/http');
-  }
-
-  return Promise.resolve(config)
-    // Wrap synchronous adapter errors and pass configuration
-    .then(adapter)
-    .then(function onFulfilled(response) {
-      // Transform response data
-      response.data = transformData(
-        response.data,
-        response.headers,
+  var adapter = config.adapter || defaults.adapter;
+
+  return adapter(config).then(function onAdapterResolution(response) {
```

Comment:
I think we better detect it at this point rather than falling back to any other implementation without errors. That way is much harder to debug, don't you think?